File: applications/mobile/shared/src/commonMain/kotlin/com/tchy/boiteacoeur/ui/theme/Theme.kt
package com.tchy.boiteacoeur.ui.theme

import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Shapes
import androidx.compose.material3.Typography
import androidx.compose.material3.darkColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

val RosePrimary = Color(0xFFFF6B8A)
val RoseSecondary = Color(0xFFE85D75)
val RoseDark = Color(0xFFB84D62)
val WineAccent = Color(0xFF8B3A4A)
val CreamHighlight = Color(0xFFFFF5F0)
val CreamMuted = Color(0xFFF5E6DC)
val PlumBackground = Color(0xFF1A0C12)
val SurfaceDark = Color(0xFF241018)
val SurfaceCard = Color(0xFF2E141C)
val TextPrimary = Color(0xFFFFF0F2)
val TextMuted = Color(0xFFC9A8B0)
val AuthGlow = Color(0x50FF6B8A)

private val DarkColors = darkColorScheme(
    primary = RosePrimary,
    onPrimary = Color(0xFF2A0A12),
    secondary = RoseSecondary,
    tertiary = WineAccent,
    background = PlumBackground,
    surface = SurfaceDark,
    surfaceVariant = SurfaceCard,
    onBackground = TextPrimary,
    onSurface = TextPrimary,
    onSurfaceVariant = TextMuted,
    outline = Color(0xFF5A3848),
)

private val AppTypography = Typography(
    headlineLarge = TextStyle(
        fontWeight = FontWeight.Bold,
        fontSize = 28.sp,
        lineHeight = 34.sp,
        letterSpacing = (-0.5).sp,
    ),
    headlineMedium = TextStyle(
        fontWeight = FontWeight.SemiBold,
        fontSize = 22.sp,
        lineHeight = 28.sp,
    ),
    titleMedium = TextStyle(
        fontWeight = FontWeight.Medium,
        fontSize = 16.sp,
        lineHeight = 22.sp,
    ),
    bodyLarge = TextStyle(
        fontSize = 16.sp,
        lineHeight = 24.sp,
    ),
    bodyMedium = TextStyle(
        fontSize = 14.sp,
        lineHeight = 20.sp,
    ),
    labelLarge = TextStyle(
        fontWeight = FontWeight.SemiBold,
        fontSize = 15.sp,
        lineHeight = 20.sp,
        letterSpacing = 0.2.sp,
    ),
)

private val AppShapes = Shapes(
    extraSmall = RoundedCornerShape(8.dp),
    small = RoundedCornerShape(12.dp),
    medium = RoundedCornerShape(16.dp),
    large = RoundedCornerShape(24.dp),
    extraLarge = RoundedCornerShape(32.dp),
)

@Composable
fun BoiteTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = DarkColors,
        typography = AppTypography,
        shapes = AppShapes,
        content = content,
    )
}

File: applications/mobile-rn/src/theme/theme.ts
export const colors = {
  rosePrimary: "#FF6B8A",
  roseSecondary: "#E85D75",
  roseDark: "#B84D62",
  wineAccent: "#8B3A4A",
  creamHighlight: "#FFF5F0",
  creamMuted: "#F5E6DC",
  plumBackground: "#1A0C12",
  surfaceDark: "#241018",
  surfaceCard: "#2E141C",
  textPrimary: "#FFF0F2",
  textMuted: "#C9A8B0",
  authGlow: "rgba(255, 107, 138, 0.16)",
  outline: "#5A3848",
  onPrimary: "#2A0A12",
  gradientTop: "#1A0812",
  gradientMid: "#1A0C12",
  gradientBottom: "#0D0208",
  success: "#5BD6A0",
  danger: "#FF7A7A",
  overlay: "rgba(10, 4, 8, 0.72)",
} as const;

export const radius = {
  xs: 8,
  sm: 12,
  md: 16,
  lg: 24,
  xl: 32,
  pill: 999,
} as const;

export const spacing = {
  xs: 4,
  sm: 8,
  md: 12,
  lg: 16,
  xl: 24,
  xxl: 32,
} as const;

export const typography = {
  headlineLarge: { fontSize: 28, lineHeight: 34, fontWeight: "700" as const, letterSpacing: -0.5 },
  headlineMedium: { fontSize: 22, lineHeight: 28, fontWeight: "600" as const },
  titleMedium: { fontSize: 16, lineHeight: 22, fontWeight: "500" as const },
  bodyLarge: { fontSize: 16, lineHeight: 24, fontWeight: "400" as const },
  bodyMedium: { fontSize: 14, lineHeight: 20, fontWeight: "400" as const },
  labelLarge: { fontSize: 15, lineHeight: 20, fontWeight: "600" as const, letterSpacing: 0.2 },
  caption: { fontSize: 12, lineHeight: 16, fontWeight: "500" as const },
} as const;

export type ThemeColors = typeof colors;
